pkg/lib.go

```diff
@@ -378,21 +367,95 @@ func GetInventoryReports(cfg *config.Application) (AccountRoutedReports, error)
 
 		// Get inventory reports for each account
 		for account, namespaces := range accountRoutesForAllNamespaces {
-			totalExpectedBatches := int(math.Ceil(float64(len(namespaces)) / float64(batchSize)))
-			for batchCount, batch := range GetNamespacesBatches(namespaces, batchSize) {
-				if batchSize > 0 {
-					log.Infof("Collecting inventory batch %d of %d for account %s", batchCount+1, totalExpectedBatches, account)
-				}
-				accountReport, err := GetInventoryReportForNamespaces(cfg, batch)
-				if err != nil {
-					return AccountRoutedReports{}, err
+			accountReport, err := GetInventoryReportForNamespaces(cfg, namespaces)
+			if err != nil {
+				return BatchedReports{}, err
+			}
+			reports[account] = accountReport
+		}
+	}
+
+	return getBatchedInventoryReports(reports, cfg.InventoryReportLimits.Namespaces), nil
+}
+
+//nolint:gocognit,funlen
+func getBatchedInventoryReports(reports AccountRoutedReports, batchSize int) BatchedReports {
+	batchedReports := BatchedReports{}
+	if batchSize <= 0 {
+		for account, report := range reports {
+			batchedReports[account] = append(batchedReports[account], report)
+		}
+		return batchedReports
+	}
+
+	log.Infof("Batching namespaces into groups of %d", batchSize)
+	for account, accountReport := range reports {
+		if len(accountReport.Namespaces) <= batchSize {
+			batchedReports[account] = append(batchedReports[account], accountReport)
+			continue
+		}
+		namespaceBatches := make([][]inventory.Namespace, 0)
+		// Construct batches of namespaces
+		for i := 0; i < len(accountReport.Namespaces); i += batchSize {
+			end := i + batchSize
+			if end > len(accountReport.Namespaces) {
+				end = len(accountReport.Namespaces)
+			}
+			namespaceBatches = append(namespaceBatches, accountReport.Namespaces[i:end])
+		}
+
+		nodeMap := make(map[string]inventory.Node)
+		for _, node := range accountReport.Nodes {
+			nodeMap[node.UID] = node
+		}
+		podMap := make(map[string]inventory.Pod)
+		for _, pod := range accountReport.Pods {
+			podMap[pod.UID] = pod
+		}
+		containersByPod := make(map[string][]inventory.Container)
```

Comment:
Is this happening after the data has been sanitized? Don't want to have an error here if we're missing data.

Reply:
no the sanitization of the report happens in (well just before) the post so shouldn't be any cause for concern